funsor/pyro/convert.py

```diff
@@ -63,21 +63,12 @@ def tensor_to_funsor(tensor, event_inputs=(), event_output=0, dtype="real"):
     assert isinstance(event_output, int) and event_output >= 0
     inputs_shape = tensor.shape[:tensor.dim() - event_output]
     output_shape = tensor.shape[tensor.dim() - event_output:]
-    dim_to_name = DIM_TO_NAME + event_inputs if event_inputs else DIM_TO_NAME
-
-    # Squeeze shape of inputs.
-    inputs = OrderedDict()
-    squeezed_shape = []
-    for dim, size in enumerate(inputs_shape):
-        if size > 1:
-            name = dim_to_name[dim - len(inputs_shape)]
-            inputs[name] = bint(size)
-            squeezed_shape.append(size)
-    squeezed_shape = torch.Size(squeezed_shape)
-    if squeezed_shape != inputs_shape:
-        tensor = tensor.reshape(squeezed_shape + output_shape)
-
-    return Tensor(tensor, inputs, dtype)
+    dim_to_name_list = DIM_TO_NAME + event_inputs if event_inputs else DIM_TO_NAME
+    dim_to_name = OrderedDict(zip(
+        range(-len(inputs_shape), 0),
+        zip(dim_to_name_list[len(dim_to_name_list) - len(inputs_shape):],
+            map(bint, inputs_shape))))
+    return to_funsor(tensor, Domain(dtype=dtype, shape=output_shape), dim_to_name)
```

Comment:
> could you point out how the `dim_to_name` kwarg is exercised by existing tests

Because of this refactoring of `funsor.pyro.convert.tensor_to_funsor`, the tests in `test/pyro/test_convert.py` that exercise `funsor.pyro.convert.tensor_to_funsor` also exercise `to_funsor`, and similarly for `to_data` via `funsor.pyro.convert.funsor_to_tensor` below.  `test/pyro/test_convert.py::test_tensor_funsor_tensor` does so most explicitly.

Reply:
Thanks for explaining!